File: 7kyu/Beginner Series 3 Sum of Numbers/Sum.java
/*
Given two integers a and b, which can be positive or negative, find the sum of all the numbers
between including them too and return it. If the two numbers are equal return a or b.

        Note: a and b are not ordered!
        Examples

        GetSum(1, 0) == 1   // 1 + 0 = 1
        GetSum(1, 2) == 3   // 1 + 2 = 3
        GetSum(0, 1) == 1   // 0 + 1 = 1
        GetSum(1, 1) == 1   // 1 Since both are same
        GetSum(-1, 0) == -1 // -1 + 0 = -1
        GetSum(-1, 2) == 2  // -1 + 0 + 1 + 2 = 2
*/


import java.util.stream.IntStream;

public class Sum {
    public int GetSum(int a, int b) {
        return IntStream.range(Math.min(a, b), Math.max(a, b) + 1).sum();
    }
}

File: 7kyu/Beginner Series 3 Sum of Numbers/Sum.cs
/*public class Sum
{
    public int GetSum(int a, int b)
    {
        int sum = 0;

        if (a > b)
            (a, b) = (b, a);

        for (int i = a; i <= b; i++)
            sum += i;

        return sum;
    }
}*/

/*using System;
using System.Linq;

public class Sum
{
    public int GetSum(int a, int b)
    {
        return Enumerable.Range(Math.Min(a, b), Math.Max(a, b) - Math.Min(a, b) + 1).Sum();
    }
}*/

using System;

public class Sum
{
    public int GetSum(int a, int b)
    {
        return (a + b) * (Math.Abs(a - b) + 1) / 2;
    }
}